src/main/java/components/ErrorSimulator.java

```diff
@@ -1,69 +1,261 @@
+
+					// Now we learn the server workers PORT
+					currentServerWorkerPort = serverResponsePacket.getPort();
+					LOG.logVerbose("Received server response. Worker thread port: " + currentServerWorkerPort);
+
+					// Now forward the initial response back to client
+					connection.forwardPacket(serverResponsePacket, clientAddress);
+					LOG.logQuiet("Forwarding initial response to client");
+				}
+				// If the packet is from the server worker
+				// We are going to forward the packet to the current client
+				else if(!isFromClient(incomingAddr, incomingPort) && isFromServerWorker(incomingAddr, incomingPort))
+				{
+					LOG.logQuiet("Received message from client. Forwarding to server.");
+					connection.forwardPacket(incomingPacket, clientAddress);
+				}
+				// If the packet is from the current client
+				// We are going to forward the packet to the server
+				else if(!isFromServerWorker(incomingAddr, incomingPort) && isFromClient(incomingAddr, incomingPort))
+				{
+					LOG.logQuiet("Received message from server. Forwarding to client.");
+					connection.forwardPacket(incomingPacket, serverAddress, currentServerWorkerPort);
+				}
+
+			} catch (SocketException sE)
+			{
+				// Socket closed exception
+				if(!connection.isClosed())
+					sE.printStackTrace();
+			}
+			catch (IOException e) {
+				e.printStackTrace();
 			}
 		}
 	}
-}
 
+	/**
+	 * Prints CLI help
+	 */
+	private static void printHelp() {
+		System.out.println("Command Line Arguments:");
+		System.out.println("ErrorSimulator [-s server_ip] [-v]");
+		System.out.println("\t[-h]: Shows Help page");
+		System.out.println("\t[-s server_address]: Where server_address is the server's IP Address. Default is localhost. Port: " + GLOBAL_CONFIG.SERVER_PORT);
+		System.out.println("\t[-v]: Sets logging to verbose");
+	}
+
+	/**
+	 * Parses the server address from the command line arguments
+	 * @param args Command Line Arguments
+	 * @return The SocketAddress to use for the connection to the server.
+	 */
+	private static InetAddress getServerAddress(String[] args)
+	{
+		// Create default socket address
+		InetAddress serverAddr = InetAddress.getLoopbackAddress();
+
+		if(args == null || args.length < 2)
+			return serverAddr;
+
+		int optionIndex = getOption(args, 's');
+
+		// Make sure server option exists
+		if(optionIndex == -1)
+		{
+			LOG.logQuiet("No server arguments provided. Using default server address.");
+			return serverAddr;
+		}
+
+		// Read next argument
+		if(optionIndex + 1 == args.length) {
+			LOG.logQuiet("Missing server address argument for option -s. Using default server address.");
+			return serverAddr;
+		}
+
+		// Create address out of parameter value
+		return new InetSocketAddress(args[optionIndex + 1], GLOBAL_CONFIG.SERVER_PORT).getAddress();
+	}
+
+	/**
+	 * @return The index of a specified option argument, or -1 if not found
+	 */
+	private static int getOption(String[] args, char option)
```

Comment:
This seems very time-consuming if we're calling this many times. A better way would be to parse the arguments into a HashMap and then call the desired option afterwards.

But it is an acceptable hack considering we are refactoring the cli

Reply:
Yeah I agree! This is not acceptable in my eyes, it was supposed to be thrown together quickly because it was necessary in a last minute change! 

If you look at the issue I raised yesterday, you'll see that one of the tasks is to generalize the command class so that it can be extended to be used by the error simulator and the server (since both use their own CLI). I think this would a much more appropriate way to handle all of this.